zenoh/src/net/routing/hat/router/token.rs

```diff
@@ -406,9 +412,26 @@ fn propagate_forget_simple_token(
                     res.expr(),
                 ),
             );
-        } else if face_hat!(face).remote_interests.values().any(|(r, o)| {
-            o.tokens() && r.as_ref().map(|r| r.matches(res)).unwrap_or(true) && !o.aggregate()
-        }) {
+        // NOTE(fuzzypixelz): We need to check that `face` is not the source Face of the token
+        // undeclaration, otherwise the undeclaration would be duplicated at the source Face.
+        //
+        // The code path that leads to `src_face` being `None` originate in:
```

Comment:
The first two lines of comment are ok to me. 
But all the rest just to explain why src_face is an option and when it is None is overkill to me. If we go that far we'll be overwhelmed with comments. And one can easily search and find places where it is called with None and why it makes sense.

Reply:
I wrote that to explain to myself, if anything :P